### ISSUE TITLE
app.json must include a JSON object

### ISSUE THREAD

**Issue by hashim-sohail** [expo/expo-cli#1392]
## Description
Getting this error in my github actions when running publish
`app.json must include a JSON object`

## Expected Behavior
Ran Expo Publish in github actions, the build should be published. 

## Observed Behavior

If there's an error message, please paste the _full terminal output and error message_ in this code block:

```
[06:12:05] Unable to find an existing Expo CLI instance for this directory, starting a new one...
[06:12:05] app.json must include a JSON object.
[06:12:05] Set EXPO_DEBUG=true in your env to view the stack trace.
```

## Environment
Github Actions

**Comment by byCedric** [comment]
Can you provide a bit more context? Right now it's hard for anyone to find the issue because we have no clue how you set everything up. An example repo would help for example.

**Comment by superandrew** [comment]
I have this problem, too.

here's my app.json file

```
{
  "expo": {
    "name": "<redacted>",
    "slug": "<redacted>",
    "privacy": "public",
    "sdkVersion": "1.0.9",
    "ios": {
      "supportsTablet": true,
      "bundleIdentifier": "<redacted>",
      "buildNumber": "1.0.7"
    },
    "android": {
      "package": "<redacted>",
      "googleServicesFile": "./google-services.json"
    },
    "platforms": [
      "ios",
      "android"
    ],
    "version": "1.0.0",
    "orientation": "portrait",
    "icon": "./assets/images/icon.png",
    "splash": {
      "image": "./assets/images/splash.png",
      "resizeMode": "contain",
      "backgroundColor": "#ffffff"
    },
    "updates": {
      "fallbackToCacheTimeout": 0
    },
    "assetBundlePatterns": [
      "**/*"
    ],
    "description": "<redacted>",
    "scheme": "<redacted>"
  }
}
```

**Comment by superandrew** [comment]
I was able to build just fine, and today I tried a tool to autoincrement the ios.buildNumber from "55" to "1.0.56" with json-version-inc. However even resetting with git the app.json file throws this error, without any help from EXPO_DEBUG set.

```
Waiting for build to complete. You can press Ctrl+C to exit.
✔ Build finished.
Successfully built standalone app: https://expo.io/artifacts/<redacted>
app.json must include a JSON object.
Error: app.json must include a JSON object.
    at parseAndValidateRootConfig (/Users/superandrew/.nvm/versions/node/v12.6.0/lib/node_modules/expo-cli/node_modules/@expo/config/src/Config.ts:96:13)
    at /Users/superandrew/.nvm/versions/node/v12.6.0/lib/node_modules/expo-cli/node_modules/@expo/config/src/Config.ts:40:31
    at Generator.throw (<anonymous>)
    at rejected (/Users/superandrew/.nvm/versions/node/v12.6.0/lib/node_modules/expo-cli/node_modules/@expo/config/build/Config.js:6:65)
```

**Comment by dereksleung** [comment]
I started getting this issue tonight when before everything was fine, and was wondering if it has to do with my expo-cli version, which is 3.5.0

**Comment by devdebajyotidas** [comment]
> I started getting this issue tonight when before everything was fine, and was wondering if it has to do with my expo-cli version, which is 3.5.0

The same is happening to me after upgrading the CLI

**Comment by wodin** [comment]
I have discovered that the behaviour of `expo` has changed wrt. parsing of command line arguments.  e.g., when running `expo publish release-channel testing` (note the invalid `release-channel` instead of `--release-channel`) the behaviour is as follows:

expo-cli@3.1.2:
```
Can't read JSON file: /private/tmp/release-channel/app.json
└─ Cause: Error: ENOENT: no such file or directory, open '/private/tmp/release-channel/app.json'
```

expo-cli@3.2.0:
```
app.json must include a JSON object.
```

So please double check the command you're running and if you believe it is correct, post a comment including the exact command you're running.

**Comment by woba-ko** [comment]
I am encountering the same issue when trying to upload a .ipa file to Appstore. Building ios/android, publishing ios/android and uploading .apk file to Play Store works without issues.

I am trying to upload a new build instead of publishing, because we updated expo from 33 to 35, And I have tried using both v3.4.1 and v3.11.3 of the CLI.

When running: 
`expo upload:ios --latest --apple-id myAppleId --apple-id-password myAppleIdPassword` 
or 
`expo upload:ios --latest myAppleId myAppleIdPassword`

it throws this error: `app.json must include a JSON object.` I have tried running the command with `EXPO_DEBUG=true` but it does not log anything. I also tried running the command with sudo to make sure that it wasn't an issue of accessibility.

As a last resort i downloaded Transporter and used that to upload instead, since it does not throw the error.

**Comment by wodin** [comment]
Hi @woba-ko 

The second command is incorrect, but the first looks OK to me. So I am not surprised that the second one gives you tge confusing `app.json must include a JSON object.` error.

What do you get if you run this from the same place as you ran `expo upload:ios ...`?

```
python -mjson.tool app.json >/dev/null && echo OK
```

**Comment by woba-ko** [comment]
@wodin I just learned that i have to run the command from within the source folder of the app. Then it does not fail. However, the script fails to upload because it stalls:

```
[09:19:20]: Going to upload updated app to App Store Connect
[09:19:20]: This might take a few minutes. Please don't interrupt the script.
[09:35:30]: [Transporter Error Output]: An error occurred while uploading the package: 1229103781.itmsp.
[09:35:30]: [Transporter Error Output]: An exception has occurred: The transport has been detected as having stalled and has been aborted.  Please try again.
```

This happens every time, without any explanation.

**Comment by exAspArk** [comment]
I tried downgrading the `expo-cli` version to 3.10.2, but now it shows this error:

```
┌──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│                                                                                                                                          │
│   There is a new version of expo-cli available (3.11.3).                                                                                 │
│   You are currently using expo-cli 3.10.2                                                                                                │
│   Install expo-cli globally using the package manager of your choice; for example: `npm install -g expo-cli` to get the latest version   │
│                                                                                                                                          │
└──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
_xdl(...).Doctor.validateLowLatencyAsync is not a function
Set EXPO_DEBUG=true in your env to view the stack trace.
```

**Comment by wodin** [comment]
Sorry, I'm not sure what would be causing that transporter error. Does that happen with the latest version of expo-cli as well?

**Comment by wodin** [comment]
Hi. You shouldn't need to downgrade.
What happens if you use the latest version of expo-cli?

**Comment by woba-ko** [comment]
@wodin yes I am using the latest version of the expo-cli: v3.11.3

**Comment by mkhoussid** [comment]
I know this is silly, but I came back to a project that I haven't touched in several months, and was running into this error. Make sure you have all the node modules installed and are executing the `expo build` command from within your client directory.

**Comment by AdarshJais** [comment]
Deploying to a standalone app on Android

and using this command to get SHAI:1 **expo fetch:android:hashes** 

and getting the error 

app.json must include a JSON object.
Set EXPO_DEBUG=true in your env to view the stack trace.

**Comment by wodin** [comment]
Hi @AdarshJais 

What do you get if you run the following from the same place as where you ran `expo fetch:android:hashes`?

```
python -mjson.tool app.json >/dev/null && echo OK
```

**Comment by fson** [comment]
We looked into this with @IjzerenHein. It seems that in previous versions of Expo CLI running a command in a folder that's not a project folder could have resulted in the error `app.json must include a JSON object` to be printed. This is no longer the case with the latest Expo CLI version (`3.15.4`).

Additionally, if `app.json` included a syntax error, there was no error message, but instead the configuration just defaulted to an empty object. This is going to be fixed by #1709.
